inbm/dispatcher-agent/tests/unit/sota/test_downloader.py

```diff
@@ -110,3 +115,89 @@ def _build_mock_repo(num_files=0):
             for i in range(0, num_files):
                 mem_repo.add("test" + str(i + 1) + ".rpm", b"0123456789")
         return mem_repo
+
+    @patch('dispatcher.sota.downloader.read_oras_token', return_value="mock_password")
+    @patch('dispatcher.sota.downloader.oras_download')
+    def test_tiberos_download_successful(self, mock_download, mock_read_token) -> None:
+        self.release_date = self.username = None
+        password = "mock_password"
+        mock_url = canonicalize_uri("https://registry-rs.internal.ledgepark.intel.com/one-intel-edge/tiberos:latest")
+
+        assert TestDownloader.sota_instance
+        TestDownloader.sota_instance.factory = SotaOsFactory(
+            MockDispatcherBroker.build_mock_dispatcher_broker(), None, []).get_os('TiberOS')
+        factory = TestDownloader.sota_instance.factory
+        assert factory
+        installer = factory.create_downloader()
+        assert isinstance(installer, TiberOSDownloader)
+
+        directory = tempfile.mkdtemp()
+        repo = DirectoryRepo(directory)
+
+        repo.add("test", b"This is a test file.")
+
+        # Calculate the SHA256 checksum
+        sha256_hash = hashlib.sha256()
+        with open(os.path.join(directory, "test"), 'rb') as file:
+            for chunk in iter(lambda: file.read(4096), b''):
+                sha256_hash.update(chunk)
+        checksum = sha256_hash.hexdigest()
+        installer._signature = checksum
+        try:
+            installer.download(self.mock_disp_broker,
+                               mock_url, repo,
+                               self.username, password, self.release_date)
+        except (SotaError, DispatcherException):
+            self.fail("raised Error unexpectedly!")
+
+        mock_read_token.assert_called_once()
+        mock_download.assert_called_once()
+
+    @patch('dispatcher.sota.downloader.read_oras_token', return_value="mock_password")
+    @patch('dispatcher.sota.downloader.oras_download')
+    def test_tiberos_download_and_signature_failed(self, mock_download, mock_read_token) -> None:
+        self.release_date = self.username = None
+        password = "mock_password"
+        mock_url = canonicalize_uri("https://registry-rs.internal.ledgepark.intel.com/one-intel-edge/tiberos:latest")
+
+        assert TestDownloader.sota_instance
+        TestDownloader.sota_instance.factory = SotaOsFactory(
+            MockDispatcherBroker.build_mock_dispatcher_broker(), None, []).get_os('TiberOS')
+        factory = TestDownloader.sota_instance.factory
+        assert factory
+        installer = factory.create_downloader()
+        assert isinstance(installer, TiberOSDownloader)
+
+        directory = tempfile.mkdtemp()
```

Comment:
use a context manager to ensure cleanup when making temporary files/directories:

```python
with temp_directory() as directory:
   # code here
```